src/globus_sdk/services/search/data.py

```diff
@@ -11,6 +11,15 @@
 SearchQueryT = t.TypeVar("SearchQueryT", bound="SearchQueryBase")
 
 
+def _format_histogram_range(
+    value: tuple[t.Any, t.Any] | MissingType,
+) -> dict[str, t.Any] | MissingType:
+    if isinstance(value, MissingType):
+        return value
```

Comment:
Similar to `None`, I find it easier to read syntax like this when dealing with singleton values:

```python
if value is None:
    return None  # Explicitly return the singleton value
```

```suggestion
    if value is MISSING:
        return MISSING
```

Reply:
Yeah, mypy seems to accept
```python
if t.TYPE_CHECKING:
    import enum

    class MissingType(enum.Enum):
        MISSING = enum.auto()

    MISSING = MissingType.MISSING
else:
    class MissingType:
        ...
```

but _boy_ I am unsure about how we might regret doing that.

Reply:
Sorry, my point was that I prefer to `return MISSING`, not `return value`! :grin:

Reply:
Ah! Yes, I also prefer
```python
if x == CONSTANT:
    return CONSTANT
```
for any values of `CONSTANT` and `==` :wink: